devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java

```diff
@@ -194,4 +217,51 @@ protected boolean containsBOM() {
         return dependencies;
     }
 
+    @Override
+    public String getProperty(String propertyName) {
+        return propertiesContent.getProperty(propertyName);
+    }
+
+    @Override
+    protected List<Dependency> getManagedDependencies() {
+        List<Dependency> constraints = new ArrayList<>();
+        readLineByLine(buildContent, new GetConstraints(constraints));
+        return constraints;
+    }
+
+    private static class GetConstraints implements Consumer<String> {
+
+        private List<Dependency> result;
+        private boolean inConstraints = false;
+        private boolean inImplementation = false;
+
+        public GetConstraints(List<Dependency> result) {
+            this.result = result;
+        }
+
+        @Override
+        public void accept(String currentLine) {
```

Comment:
This does not appear to catch `enforcedPlatform` or `platform`. This is inconsistent with Maven's `model.getDependencyManagement()`. Note sure whether it's a big deal though.

Reply:
done.

I am just dig a little more to not use string and delegate parse of model to gradle lib.
I see how eclipse do it:
https://github.com/eclipse/buildship/blob/master/org.eclipse.buildship.kotlin/build.gradle
https://github.com/eclipse/buildship/blob/master/org.eclipse.buildship.kotlin/src/main/java/org/eclipse/buildship/kotlin/internal/GradleKotlinScriptTemplateProvider.java#L80

And with ProjectConnection we can load the groovy model  too, but I don't know the limitation and we need all gradle core as dependency. Not sur we want this....
Do you want I create an issue for this @aloubyansky ?

Reply:
Ideally, yes, we want to operate on a representation of the config model, as in Maven equivalent. Relying on string parsing is kind of fragile, since e.g. the formatting or composition of the config could vary. So, yes, I think it's worth creating an issue for that. Thanks @Dufgui